src/main/java/com/snowflake/kafka/connector/internal/streaming/DirectTopicPartitionChannel.java

```diff
@@ -1163,189 +922,38 @@ private SinkRecord getSnowflakeSinkRecordFromKafkaRecord(final SinkRecord kafkaS
     return snowflakeRecord;
   }
 
-  /**
-   * Get Approximate size of Sink Record which we get from Kafka. This is useful to find out how
-   * much data(records) we have buffered per channel/partition.
-   *
-   * <p>This is an approximate size since there is no API available to find out size of record.
-   *
-   * <p>We first serialize the incoming kafka record into a Json format and find estimate size.
-   *
-   * <p>Please note, the size we calculate here is not accurate and doesnt match with actual size of
-   * Kafka record which we buffer in memory. (Kafka Sink Record has lot of other metadata
-   * information which is discarded when we calculate the size of Json Record)
-   *
-   * <p>We also do the same processing just before calling insertRows API for the buffered rows.
-   *
-   * <p>Downside of this calculation is we might try to buffer more records but we could be close to
-   * JVM memory getting full
-   *
-   * @param kafkaSinkRecord sink record received as is from Kafka (With connector specific converter
-   *     being invoked)
-   * @return Approximate long size of record in bytes. 0 if record is broken
-   */
-  protected long getApproxSizeOfRecordInBytes(SinkRecord kafkaSinkRecord) {
```

Comment:
oh nice were removing this! we actually make this approximation twice per record, so hopefully this will speed up ingestion to not make this pass through